discoverd/server/handler.go

```diff
@@ -108,15 +110,25 @@ func proxyWhitelisted(r *http.Request) bool {
 
 func (h *Handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if h.Shutdown.Load().(bool) {
-		hh.Error(w, ErrShutdown)
+		hh.ServiceUnavailableError(w, "discoverd: shutting down")
 		return
 	}
 	// If running in proxy mode then redirect requests to a random peer
-	if h.Proxy.Load().(bool) && !proxyWhitelisted(r) {
-		// TODO(jpg): Should configuring the peer in proxy mode with no peers be impossible?
-		host := h.Peers[rand.Intn(len(h.Peers))]
-		redirectToHost(w, r, host)
-		return
+	if h.Proxy.Load().(bool) {
+		if !proxyWhitelisted(r) {
+			// TODO(jpg): Should configuring the peer in proxy mode with no peers be impossible?
+			host := h.Peers[rand.Intn(len(h.Peers))]
+			redirectToHost(w, r, host)
+			return
+		}
+	} else {
+		// Send current peer list and index to the client so it can keep the list of
+		// peers in sync with the cluster.
+		peers, err := h.Store.GetPeers()
+		if err == nil {
+			w.Header().Set("Discoverd-Current-Peers", strings.Join(peers, ","))
+		}
+		w.Header().Set("Discoverd-Current-Index", strconv.FormatUint(h.Store.LastIndex(), 10))
```

Comment:
Should we drop `-Current` from these so that it's just `Discoverd-Peers` and `Discoverd-Index`?